### DISCUSSION TITLE
Error during execution

### DISCUSSION THREAD

**Discussion by exptjs** [sequelize/umzug#670] [Q&A]
[Example](https://github.com/exptjs/test-umzug).

When executing `ts-node migrate --help` errors:
```
(node:9588) [SEQUELIZE0030] DeprecationWarning: Do not use sequelize.getQueryInterface(). Use sequelize.queryInterface instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:9588) [SEQUELIZE0029] DeprecationWarning: Do not use sequelize.isDefined(). Use sequelize.models.hasByName instead.
```
But the documentation says that it is necessary to use `context: _sequelize.gt queryinterface()`. Why is it suggested to use `sequelize.queryInterface`?
How should I use `sequelize.models.hasByName`? I didn't find anything in the documentation.

`ts-node migrate create --name tableA.ts` errors:
```
(node:3948) [SEQUELIZE0030] DeprecationWarning: Do not use sequelize.getQueryInterface(). Use sequelize.queryInterface instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:3948) [SEQUELIZE0029] DeprecationWarning: Do not use sequelize.isDefined(). Use sequelize.models.hasByName instead.

Error: Couldn't infer a directory to generate migration file in. Pass --folder explicitly
```
Why is the configuration ignored in this case? `migrations: {glob: ['migrations/*.{ts,sql}', {cwd: __dirname}]}`

When running `ts-node migrate create --name tableA.ts --folder migrations` an error occurs, the migration file is created, but the command does not end. Can you tell me what this error means?
```
Error: SASL: SCRAM-SERVER-FIRST-MESSAGE: client password must be a string
```
I did not find any information in the documentation to specify the user or database password for performing migrations. Tell me, it needs to be transferred to `const _sequelize = new Sequelize({
  dialect: 'postgres',
  logging: console.log
});
`?

**Comment by exptjs** [comment]
@mmkal can I get an answer? These errors do not allow me to use umzug

**Comment by WikiRik** [comment]
Comenting here from the sequelize perspective and not umzug specifically. The DeprecationWarnings that you are getting are from sequelize v7 alphas, and umzug officially does not support that yet. That's why you see those. But they're only warnings and not the casue of any issues. 

For the Sequelize constructor in the newest alpha, check out this documentation; https://sequelize.org/docs/v7/databases/postgres/
That explains how to specify the connection options, such as username and password

**Reply by exptjs** [reply]
@WikiRik The [sequelize-cli](https://sequelize.org/docs/v7/cli/) documentation says that `umzug` can be used, because `sequelize-cli` is not compatible with `sequelize 7 alpha`. And the `umzug` documentation does not say that it is not compatible with `sequelize 7 alpha`.
Can you tell me when to expect support for `umzug` with `sequelize 7 alpha` and when will the stable version of `sequelize 7` be released? I have been using `sequelize` for a long time and at the moment typescript and `umzug` support is needed, so far I see a better solution, because other solutions are completely unsuitable.
